feat: Developer Guide - GUI - MySQL Workbench (#15366)
pingcap/docs: develop/dev-guide-gui-mysql-workbench.md

2. Click **Connect** in the upper-right corner. A connection dialog is displayed.

3. Ensure the configurations in the connection dialog match your operating environment.

    - **Endpoint Type** is set to `Public`.
    - **Connect With** is set to `General`.
    - **Operating System** matches your environment.

4. Click **Create password** to create a random password.

    > **Tip:**
    >
    > If you have created a password before, you can either use the original password or click **Reset password** to generate a new one.

5. Launch MySQL Workbench and click the **+** in the right side of **MySQL Connections**.

    ![MySQL Workbench: add new connection](/media/develop/navicat-add-new-connection.png)

6. In the setup panel, configure the following connection parameters:

    - **Connection Name**: give this connection a meaningful name.
    - **Hostname**: enter the `host` parameter from the TiDB Cloud connection dialog.
    - **Port**: enter the `port` parameter from the TiDB Cloud connection dialog.
    - **Username**: enter the `user` parameter from the TiDB Cloud connection dialog.

>     - **Password**: click **Store in Keychain ...**, enter the password of the TiDB Serverless cluster, then click **OK** to store the password.

Reviewer: It Keychain available on all platforms, including Windows?

Reviewer 2: It is: https://dev.mysql.com/doc/workbench/en/wb-mysql-connections-vault.html

>         ![MySQL Workbench: store password in keychain](/media/develop/mysql-workbench-store-password-in-keychain.png)

Reviewer 3: ```suggestion
        ![MySQL Workbench: store the password of TiDB Serverless in keychain](/media/develop/mysql-workbench-store-password-in-keychain.png)
```

2. Click **Connect** in the upper-right corner. A connection dialog is displayed.

3. Click **Allow Access from Anywhere**.

    For more details about how to obtain the connection string, refer to [TiDB Dedicated standard connection](https://docs.pingcap.com/tidbcloud/connect-via-standard-connection).

4. Launch MySQL Workbench and click the **+** in the right side of **MySQL Connections**.

    ![MySQL Workbench: add new connection](/media/develop/navicat-add-new-connection.png)

5. In the setup panel, configure the following connection parameters:

    - **Connection Name**: give this connection a meaningful name.
    - **Hostname**: enter the `host` parameter from the TiDB Cloud connection dialog.
    - **Port**: enter the `port` parameter from the TiDB Cloud connection dialog.
    - **Username**: enter the `user` parameter from the TiDB Cloud connection dialog.
    - **Password**: click **Store in Keychain ...**, enter the password of the TiDB Dedicated cluster, then click **OK** to store the password.

        ![MySQL Workbench: store Dedicated Tier password in keychain](/media/develop/mysql-workbench-store-dedicated-password-in-keychain.png)

>     ![MySQL Workbench: configure connection parameters panel for TiDB Dedicated](/media/develop/mysql-workbench-connection-config-dedicated-parameters.png)

Reviewer 3: ```suggestion
   The following figure shows an example of the connection parameters:

    ![MySQL Workbench: configure connection settings for TiDB Dedicated](/media/develop/mysql-workbench-connection-config-dedicated-parameters.png)
```

6. Click **Test Connection**, if the connection test is successful, you can see the **Successfully made the MySQL connection** message. Click **OK** to save the connection configuration.

</div>
<div label="TiDB Self-Hosted">

1. Launch MySQL Workbench and click the **+** in the right side of **MySQL Connections**.

    ![MySQL Workbench: add new connection](/media/develop/navicat-add-new-connection.png)

> 2. In the setup panel, configure the following connection parameters:

Reviewer 3: ```suggestion
2. In the **Setup New Connection** dialog, configure the following connection parameters:
```

    - **Connection Name**: give this connection a meaningful name.
    - **Hostname**: enter the IP address or domain name of your TiDB Self-Hosted cluster.
    - **Port**: enter the port number of your TiDB Self-Hosted cluster.
    - **Username**: enter the username to use to connect to your TiDB.
    - **Password**: click **Store in Keychain ...**, enter the password to use to connect to your TiDB, then click **OK** to store the password.

>         ![MySQL Workbench: store self-hosted password in keychain](/media/develop/mysql-workbench-store-self-hosted-password-in-keychain.png)

Reviewer 3: ```suggestion
        ![MySQL Workbench: store the password of TiDB Self-Hosted in keychain](/media/develop/mysql-workbench-store-self-hosted-password-in-keychain.png)
```

>     ![MySQL Workbench: configure connection parameters panel for self-hosted TiDB](/media/develop/mysql-workbench-connection-config-self-hosted-parameters.png)

Reviewer 3: ```suggestion

   The following figure shows an example of the connection parameters:

    ![MySQL Workbench: configure connection settings for TiDB Self-Hosted](/media/develop/mysql-workbench-connection-config-self-hosted-parameters.png)
```

> 3. **Test Connection** to validate the connection to the self-hosted TiDB.
> 
> 4. If the connection test is successful, you can see the **Connection Successful** message. Click **Save** to save the connection configuration.

Reviewer 3: ```suggestion
3. Click **Test Connection** to validate the connection to the TiDB Self-Hosted cluster.

4. If the connection test is successful, you can see the **Successfully made the MySQL connection** message. Click **OK** to save the connection configuration.
```

</div>
</SimpleTab>

## Next steps

- Learn more usage of MySQL Workbench from [the documentation of MySQL Workbench](https://dev.mysql.com/doc/workbench/en/).
- Learn the best practices for TiDB application development with the chapters in the [Developer guide](/develop/dev-guide-overview.md), such as [Insert data](/develop/dev-guide-insert-data.md), [Update data](/develop/dev-guide-update-data.md), [Delete data](/develop/dev-guide-delete-data.md), [Single table reading](/develop/dev-guide-get-data-from-single-table.md), [Transactions](/develop/dev-guide-transaction-overview.md), and [SQL performance optimization](/develop/dev-guide-optimize-sql-overview.md).
- Learn through the professional [TiDB developer courses](https://www.pingcap.com/education/) and earn [TiDB certifications](https://www.pingcap.com/education/certification/) after passing the exam.

## Need help?

> Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), or [create a support ticket](https://support.pingcap.com/).

Reviewer 2: ```suggestion
Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), [Slack](https://slack.tidb.io/invite?team=tidb-community&channel=everyone&ref=pingcap) or [create a support ticket](https://support.pingcap.com/).
```

> Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), [Slack](https://slack.tidb.io/invite?team=tidb-community&channel=everyone&ref=pingcap) or [create a support ticket](https://support.pingcap.com/).

Reviewer 2: ```suggestion
Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), [Slack](https://slack.tidb.io/invite?team=tidb-community&channel=everyone&ref=pingcap) or [create a support ticket](/support.md).
```

> Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), [Slack](https://slack.tidb.io/invite?team=tidb-community&channel=everyone&ref=pingcap) or [create a support ticket](https://support.pingcap.com/).

Reviewer 3: ```suggestion
Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), [Slack](https://slack.tidb.io/invite?team=tidb-community&channel=everyone&ref=pingcap), or [create a support ticket](/support.md).
```

    - **Connect With** is set to `General`.
    - **Operating System** matches your environment.

4. Click **Create password** to create a random password.

    > **Tip:**
    >
    > If you have created a password before, you can either use the original password or click **Reset password** to generate a new one.

5. Launch MySQL Workbench and click **+** near the **MySQL Connections** title.

    ![MySQL Workbench: add new connection](/media/develop/navicat-add-new-connection.png)

6. In the **Setup New Connection** dialog, configure the following connection parameters:

    - **Connection Name**: give this connection a meaningful name.
    - **Hostname**: enter the `host` parameter from the TiDB Cloud connection dialog.
    - **Port**: enter the `port` parameter from the TiDB Cloud connection dialog.
    - **Username**: enter the `user` parameter from the TiDB Cloud connection dialog.
    - **Password**: click **Store in Keychain ...**, enter the password of the TiDB Serverless cluster, and then click **OK** to store the password.

        ![MySQL Workbench: store the password of TiDB Serverless in keychain](/media/develop/mysql-workbench-store-password-in-keychain.png)

>    The following figure shows an example of the connection parameters:

Reviewer 3: ```suggestion
    The following figure shows an example of the connection parameters:
```

2. Click **Connect** in the upper-right corner. A connection dialog is displayed.

3. Click **Allow Access from Anywhere**.

    For more details about how to obtain the connection string, refer to [TiDB Dedicated standard connection](https://docs.pingcap.com/tidbcloud/connect-via-standard-connection).

4. Launch MySQL Workbench and click **+** near the **MySQL Connections** title.

    ![MySQL Workbench: add new connection](/media/develop/navicat-add-new-connection.png)

5. In the **Setup New Connection** dialog, configure the following connection parameters:

    - **Connection Name**: give this connection a meaningful name.
    - **Hostname**: enter the `host` parameter from the TiDB Cloud connection dialog.
    - **Port**: enter the `port` parameter from the TiDB Cloud connection dialog.
    - **Username**: enter the `user` parameter from the TiDB Cloud connection dialog.
    - **Password**: click **Store in Keychain ...**, enter the password of the TiDB Dedicated cluster, and then click **OK** to store the password.

        ![MySQL Workbench: store the password of TiDB Dedicated in keychain](/media/develop/mysql-workbench-store-dedicated-password-in-keychain.png)

>    The following figure shows an example of the connection parameters:

Reviewer 3: ```suggestion
    The following figure shows an example of the connection parameters:
```

6. Click **Test Connection** to validate the connection to the TiDB Dedicated cluster.

7. If the connection test is successful, you can see the **Successfully made the MySQL connection** message. Click **OK** to save the connection configuration.

</div>
<div label="TiDB Self-Hosted">

1. Launch MySQL Workbench and click **+** near the **MySQL Connections** title.

    ![MySQL Workbench: add new connection](/media/develop/navicat-add-new-connection.png)

2. In the **Setup New Connection** dialog, configure the following connection parameters:

    - **Connection Name**: give this connection a meaningful name.
    - **Hostname**: enter the IP address or domain name of your TiDB Self-Hosted cluster.
    - **Port**: enter the port number of your TiDB Self-Hosted cluster.
    - **Username**: enter the username to use to connect to your TiDB.
    - **Password**: click **Store in Keychain ...**, enter the password to use to connect to your TiDB cluster, and then click **OK** to store the password.

        ![MySQL Workbench: store the password of TiDB Self-Hosted in keychain](/media/develop/mysql-workbench-store-self-hosted-password-in-keychain.png)

> 

Reviewer 3: ```suggestion
```

>    The following figure shows an example of the connection parameters:

Reviewer 3: ```suggestion
    The following figure shows an example of the connection parameters:
```

</div>
</SimpleTab>

## Next steps

- Learn more usage of MySQL Workbench from [the documentation of MySQL Workbench](https://dev.mysql.com/doc/workbench/en/).
- Learn the best practices for TiDB application development with the chapters in the [Developer guide](/develop/dev-guide-overview.md), such as [Insert data](/develop/dev-guide-insert-data.md), [Update data](/develop/dev-guide-update-data.md), [Delete data](/develop/dev-guide-delete-data.md), [Single table reading](/develop/dev-guide-get-data-from-single-table.md), [Transactions](/develop/dev-guide-transaction-overview.md), and [SQL performance optimization](/develop/dev-guide-optimize-sql-overview.md).
- Learn through the professional [TiDB developer courses](https://www.pingcap.com/education/) and earn [TiDB certifications](https://www.pingcap.com/education/certification/) after passing the exam.

## Need help?

<CustomContent platform="tidb">

Ask questions on the [Discord](https://discord.gg/DQZ2dy3cuc?utm_source=doc), or [create a support ticket](/support.md).

</CustomContent>

<CustomContent platform="tidb-cloud">

> Ask questions on the [Discord](https://discord.gg/DQZ2dy3cuc?utm_source=doc), or [create a support ticket](https://support.pingcap.com/).

Reviewer 3: ```suggestion
Ask questions on the [Discord](https://discord.gg/DQZ2dy3cuc?utm_source=doc), or [create a support ticket](/tidb-cloud/tidb-cloud-support.md).
```
